src/quick-lint-js/fe/parse-statement.cpp

```diff
@@ -3977,6 +3977,11 @@ void Parser::parse_and_visit_if(Parse_Visitor_Base &v) {
     parse_and_visit_body();
     break;
 
+  case Token_Type::semicolon:
+    if (!this->options_.typescript) {
+      parse_and_visit_body();
+      break;
+    }
```

Comment:
Please explicitly fall through:
```suggestion
  case Token_Type::semicolon:
    if (!this->options_.typescript) {
      parse_and_visit_body();
      break;
    }
    [[fallthrough]];
```

But I think it's better copy-paste the code below and invert the `if`:
```suggestion
  case Token_Type::semicolon:
    if (this->options_.typescript) {
      this->diag_reporter_->report(Diag_Missing_Body_For_If_Statement{
          .expected_body =
              Source_Code_Span::unit(this->lexer_.end_of_previous_token()),
      });
    } else {
      parse_and_visit_body();
    }
    break;
```